### PR TITLE
feat: [draft] add TelemetryLoggerMiddleware to runtime

### LINKED ISSUE #1639: LUIS and Dialog telemetry in Sample Bot Project
**Is your feature request related to a problem? Please describe.**
Telemetry support for Bot created using Composer UI.

**Describe the solution you'd like**
Application insights telemetry support for Recognizer(LUIS) and Dialogs in Bot project

**Additional context**
This will help in monitoring scenarios and for utilizing Power BI report template in Virtual assistant template(https://microsoft.github.io/botframework-solutions/solution-accelerators/tutorials/view-analytics/1-intro/).

### PR DESCRIPTION
**Pull request by zidaneymar**
## Description

1. fixes #1639
2. support #2240 (not sure, just follow the documentation here: https://docs.microsoft.com/en-us/azure/bot-service/bot-service-channel-directline-extension-net-bot?view=azure-bot-service-4.0)
  
image

### CODE REVIEW

File: BotProject/Templates/CSharp/Program.cs [hunk 1]
```diff
@@ -42,11 +42,6 @@ public static void Main(string[] args)
                     builder.AddJsonFile("appsettings.deployment.json", optional: true, reloadOnChange: true);
                 }
 
-                if (!env.IsDevelopment())
```
Comment (luhan2017) [inline BotProject/Templates/CSharp/Program.cs:45]: why are you removing this?

Comment (zidaneymar) [reply BotProject/Templates/CSharp/Program.cs:45]: currently we don't have an user secret inside the botruntime, app password and luis authoring key are stored in file(appsettings)

Comment (zidaneymar) [reply BotProject/Templates/CSharp/Program.cs:45]: > why are you removing this?

currently we don't have an user secret inside the botruntime, app password and luis authoring key are stored in file(appsettings)

File: BotProject/Templates/CSharp/BotProject.csproj [hunk 2]
```diff
@@ -31,6 +31,7 @@
     <PackageReference Include="Microsoft.Bot.Builder.Integration.ApplicationInsights.Core" Version="4.8.0" />
     <PackageReference Include="Microsoft.Bot.Builder.Integration.AspNet.Core" Version="4.8.0" />
     <PackageReference Include="Microsoft.Bot.Builder.Dialogs" Version="4.8.0" />
+    <PackageReference Include="Microsoft.Bot.Builder.StreamingExtensions" Version="4.6.0-preview-191021-1" />
```
Comment (cwhitten) [inline BotProject/Templates/CSharp/BotProject.csproj:34]: why is this a 4.6.0 preview package? shouldn't it be 4.8.0?

**Pull request closed without merging**